### PR TITLE
Use catalog service for course info.

### PR DESCRIPTION
**Pull request by mjfrey**
Migrate single course coupons to pull course info from catalog service rather than LMS.

### CODE REVIEW

**Comment by mjfrey** [comment]
@vkaracic @marjev @mattdrayer Please review.

File: ecommerce/courses/utils.py [hunk 1]
```diff
@@ -23,14 +20,15 @@ def mode_for_seat(product):
...
 
 
-def get_course_info_from_lms(course_key):
-    """ Get course information from LMS via the course api and cache """
-    api = EdxRestApiClient(get_lms_url('api/courses/v1/'))
-    cache_key = 'courses_api_detail_{}'.format(course_key)
+def get_course_info_from_catalog(site, course_key):
+    """ Get course information from catalog service and cache """
+    api = site.siteconfiguration.course_catalog_api_client
+    partner = site.siteconfiguration.partner.short_code
```
Comment (mattdrayer) [inline ecommerce/courses/utils.py]: I'd rename this to `partner_short_code`

File: ecommerce/extensions/basket/views.py [hunk 2]
```diff
@@ -106,12 +106,13 @@ def get_context_data(self, **kwargs):
...
-                course = get_course_info_from_lms(course_key)
-                image_url = get_lms_url(course['media']['course_image']['uri'])
-                short_description = course['short_description']
-                course_name = course['name']
+                course = get_course_info_from_catalog(self.request.site, course_key)
+                description = course['short_description']
+                image_url = course['image']['src']
+                short_description = description if description else ''
+                course_name = course['title']
             except (ConnectionError, SlumberBaseException, Timeout):
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/views.py:114]: The try/except should cover only what needs to be protected -- in this case `get_course_info_from_catalog`

File: ecommerce/courses/utils.py [hunk 3]
```diff
@@ -23,14 +20,15 @@ def mode_for_seat(product):
...
+    cache_key = 'courses_api_detail_{}{}'.format(course_key, partner)
     cache_hash = hashlib.md5(cache_key).hexdigest()
     course = cache.get(cache_hash)
     if not course:  # pragma: no cover
-        course = api.courses(course_key).get()
+        course = api.course_runs(course_key).get(partner=partner)
```
Comment (mattdrayer) [inline ecommerce/courses/utils.py]: Is this a course or a course run?  I see we are calling the variable `course` but loading the data from the `course_runs` api

**Comment by mattdrayer** [comment]
Just a couple comments @mjfrey -- overall the changes LGTM -- nice work!  👍

File: ecommerce/courses/utils.py (hunk 3 above)
Comment (mjfrey) [reply ecommerce/courses/utils.py]: we get the course_run for the course.  That is what is purchasable.

Comment (mattdrayer) [reply ecommerce/courses/utils.py]: Right, so maybe we should rename the course variable to course_run here

**Comment by vkaracic** [comment]
:+1:

**Comment by marjev** [comment]
👍

**Pull request merged**